File: s3-inf-ppo/laboratoria/lab10/java/src/Notebook/Menu.java
package Notebook;

import Notebook.Entries.DrankBeer;
import Notebook.Entries.WatchedMovie;
import Notebook.Interfaces.NotebookEntry;

import java.util.Scanner;

final public class Menu {

    private Notebook notebook;

    public Menu(Notebook notebook) {

        this.notebook = notebook;
    }

    public void run() {
        boolean running = true;
        Scanner scanner = new Scanner(System.in);

        while(running) {
            displayMenu();

            write("Wybierz opcję: ");
            switch(scanner.nextLine()) {
                case "1": {
                    write();
                    comment("zanotowano:");
                    for(NotebookEntry item : notebook.getAll()) {
                        write(item.getSlug());
                    }
                    break;
                }
                case "2": {
                    write("Podaj tytuł filmu: ");
                    String title = scanner.nextLine();
                    WatchedMovie movie = new WatchedMovie(title);
                    notebook.save(movie);
                    break;
                }
                case "3": {
                    write("Podaj nazwę piwa: ");
                    String name = scanner.nextLine();
                    write("Podaj nazwę browaru: ");
                    String brewery = scanner.nextLine();
                    DrankBeer beer = new DrankBeer(name, brewery);
                    notebook.save(beer);
                    break;
                }
                case "x": {
                    running = false;
                    break;
                }
                default: {
                    break;
                }
            }
        }
    }

    private void displayMenu() {
        write();
        comment("notatek: ");
        write("[1] wypisz notatki");
        write("[2] zanotuj obejrzany film");
        write("[3] zanotuj wypite piwo");
        write("[x] wyjdź");
    }

    private void write() {
        System.out.println();
    }

    private void write(String $line) {
        System.out.println($line);
    }

    private void comment(String line) {
        write(line);
    }

}

File: s3-inf-ppo/laboratoria/lab10/php/src/Menu.php
<?php

namespace PPO\Notebook;

use PPO\Notebook\Entries\DrankBeer;
use PPO\Notebook\Entries\WatchedMovie;

final class Menu {

	private $notebook;

	public function __construct(Notebook $notebook) {
		$this->notebook = $notebook;
	}

	public function run(): void {
		$running = true;

		while($running) {
			$this->displayMenu();

			switch(readline("Wybierz opcję: ")) {
				case 1: {
					$this->write();
					$this->write("zanotowano:", true);
					foreach($this->notebook->getAll() as $entry) {
						$this->write($entry->getSlug());
					}
					break;
				}
				case 2: {
					$title = readline("Podaj tytuł filmu: ");
					$movie = new WatchedMovie($title);
					$this->notebook->save($movie);
					break;
				}
				case 3: {
					$name = readline("Podaj nazwę piwa: ");
					$brewery = readline("Podaj nazwę browaru: ");
					$beer = new DrankBeer($name, $brewery);
					$this->notebook->save($beer);
					break;
				}
				case "x": {
					$running = false;
					break;
				}
				default: {
					break;
				}
			}
		}
	}

	private function displayMenu(): void {
		$this->write();
		$this->write("notatek: {$this->notebook->countEntries()}", true);
		$this->write("[1] wypisz notatki");
		$this->write("[2] zanotuj obejrzany film");
		$this->write("[3] zanotuj wypite piwo");
		$this->write("[x] wyjdź");
	}

	private function write(string $line = "", $comment = false): void {
		if($comment) echo "::: ";
		echo $line . PHP_EOL;
	}

}